lib/reverse_sentence.rb

```diff
@@ -1,6 +1,56 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n), where n is the length of the sentence
+# Space complexity: O(1), as all transformations are done in place, no additional space is needed
+
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  unless my_sentence == nil
+    temp_sentence = string_reverse(my_sentence)
+    reverse_words(temp_sentence)
+  end
+  return my_sentence
 end
+
+def reverse_words(my_words)
+  first_char = 0
+  last_char = 0
+  i = 0
+  unless my_words == nil
+    length = my_words.length
+    while i < length
+      while my_words[i] == " "
+        i += 1
+      end
+      first_char = i
+      while i < length && my_words[i] != " "
+        i += 1
+      end
+      last_char = i - 1
+      while last_char > first_char
```

Comment:
I would suggest the part where you reverse one work, would make a good helper method for this helper.  You could also have methods to find last letter in current word etc.  That might make this method less lengthy.